nautobot-app/{{ cookiecutter.project_slug }}/tasks.py

```diff
@@ -640,10 +640,11 @@ def help_task(context):
 )
 def generate_release_notes(context, version=""):
     """Generate Release Notes using Towncrier."""
-    command = "env DJANGO_SETTINGS_MODULE=nautobot.core.settings towncrier build"
+    command = "poetry run towncrier build"
```

Comment:
When we do a towncrier build for core we typically set the version manually with `towncrier build --version x.y.z`. The version in the `pyproject.toml` doesn't typically match the version being released.

Reply:
> When we do a towncrier build for core we typically set the version manually with `towncrier build --version x.y.z`. The version in the `pyproject.toml` doesn't typically match the version being released.

That's kinda weird. The original source for this doc was the core release checklist ;)

For apps it totally should match as there's an explicit step to bump it up in the checklist.

Reply:
We call out in the checklist to run `invoke generate-release-notes --version 1.4.2` but if someone doesn't provide the version, it now falls back to the value provided by `poetry version -s` which is what's in the pyproject file.

Reply:
This only works when Nautobot >= v2.3.1 is installed btw. Otherwise you get this:

```
(nautobot-dev-example-py3.12) gary@gary-work:~/github/nautobot/nautobot-app-dev-example (u/gas-python3.12)$ towncrier build --version 1.6.6
Loading template...
Finding news fragments...
Rendering news fragments...
ERROR: tried to import nautobot_dev_example, but ran into this error: No module named 'structlog'
Traceback (most recent call last):
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/towncrier/_project.py", line 21, in _get_package
    module = import_module(package)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/home/gary/github/nautobot/nautobot-app-dev-example/nautobot_dev_example/__init__.py", line 6, in <module>
    from nautobot.apps import NautobotAppConfig
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/apps/__init__.py", line 3, in <module>
    from nautobot.core.settings_funcs import ConstanceConfigItem
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/core/settings_funcs.py", line 8, in <module>
    import structlog
ModuleNotFoundError: No module named 'structlog'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/bin/towncrier", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/towncrier/build.py", line 123, in _main
    return __main(
           ^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/towncrier/build.py", line 202, in __main
    project_name = get_project_name(
                   ^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/towncrier/_project.py", line 69, in get_project_name
    module = _get_package(package_dir, package)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/towncrier/_project.py", line 28, in _get_package
    module = import_module(package)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/home/gary/github/nautobot/nautobot-app-dev-example/nautobot_dev_example/__init__.py", line 6, in <module>
    from nautobot.apps import NautobotAppConfig
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/apps/__init__.py", line 3, in <module>
    from nautobot.core.settings_funcs import ConstanceConfigItem
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/core/settings_funcs.py", line 8, in <module>
    import structlog
ModuleNotFoundError: No module named 'structlog'
```

Related:

```
>>> import nautobot_dev_example
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gary/github/nautobot/nautobot-app-dev-example/nautobot_dev_example/__init__.py", line 6, in <module>
    from nautobot.apps import NautobotAppConfig
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/apps/__init__.py", line 3, in <module>
    from nautobot.core.settings_funcs import ConstanceConfigItem
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-dev-example-xnt8oPZM-py3.12/lib/python3.12/site-packages/nautobot/core/settings_funcs.py", line 8, in <module>
    import structlog
ModuleNotFoundError: No module named 'structlog'
```

Reply:
It's working for me with the unmodified lockfile and py3.11 on both `develop` (nautobot 2.1.4) and your `u/gas-python3.12` branch (nautobot 2.3.1).

I cannot test right now with py3.12 but I suspect you'll need to update towncrier from 23.6.0 (in poetry.lock) which doesn't support py3.12.

Reply:
Looks like it was my poetry install that was broken